x-pack/plugins/profiling_data_access/server/services/fetch_flamechart/index.ts

```diff
@@ -16,40 +17,70 @@ export interface FetchFlamechartParams {
   rangeFromMs: number;
   rangeToMs: number;
   kuery: string;
+  useLegacyFlamegraphAPI?: boolean;
 }
 
+const targetSampleSize = 20000; // minimum number of samples to get statistically sound results
+
 export function createFetchFlamechart({ createProfilingEsClient }: RegisterServicesParams) {
-  return async ({ esClient, rangeFromMs, rangeToMs, kuery }: FetchFlamechartParams) => {
+  return async ({
+    esClient,
+    rangeFromMs,
+    rangeToMs,
+    kuery,
+    useLegacyFlamegraphAPI = false,
+  }: FetchFlamechartParams) => {
     const rangeFromSecs = rangeFromMs / 1000;
     const rangeToSecs = rangeToMs / 1000;
 
     const profilingEsClient = createProfilingEsClient({ esClient });
-    const targetSampleSize = 20000; // minimum number of samples to get statistically sound results
 
     const totalSeconds = rangeToSecs - rangeFromSecs;
+    // Use legacy stack traces API to fetch the flamegraph
+    if (useLegacyFlamegraphAPI) {
+      const { events, stackTraces, executables, stackFrames, totalFrames, samplingRate } =
+        await searchStackTraces({
+          client: profilingEsClient,
+          rangeFrom: rangeFromSecs,
+          rangeTo: rangeToSecs,
+          kuery,
+          sampleSize: targetSampleSize,
+        });
+
+      return await withProfilingSpan('create_flamegraph', async () => {
+        const tree = createCalleeTree(
+          events,
+          stackTraces,
+          stackFrames,
+          executables,
+          totalFrames,
+          samplingRate
+        );
 
-    const { events, stackTraces, executables, stackFrames, totalFrames, samplingRate } =
-      await searchStackTraces({
-        client: profilingEsClient,
-        rangeFrom: rangeFromSecs,
-        rangeTo: rangeToSecs,
-        kuery,
-        sampleSize: targetSampleSize,
+        return createBaseFlameGraph(tree, samplingRate, totalSeconds);
       });
+    }
 
-    const flamegraph = await withProfilingSpan('create_flamegraph', async () => {
-      const tree = createCalleeTree(
-        events,
-        stackTraces,
-        stackFrames,
-        executables,
-        totalFrames,
-        samplingRate
-      );
-
-      return createBaseFlameGraph(tree, samplingRate, totalSeconds);
+    const flamegraph = await profilingEsClient.profilingFlamegraph({
+      query: {
+        bool: {
+          filter: [
+            ...kqlQuery(kuery),
+            {
+              range: {
+                ['@timestamp']: {
+                  gte: String(rangeFromSecs),
+                  lt: String(rangeToSecs),
+                  format: 'epoch_second',
+                  boost: 1.0,
```

Comment:
Nit: The [default value for `boost`](https://www.elastic.co/guide/en/elasticsearch/reference/8.10/query-dsl-range-query.html#range-query-field-params) is anyway 1.0. Besides we're in a [filter context](https://www.elastic.co/guide/en/elasticsearch/reference/8.10/query-filter-context.html#filter-context) where scores won't matter anyway. So while the parameter is harmless, we might just as well remove it.